### PR TITLE
[Audio 3/?] Set up building samples

### PR DESCRIPTION
**Pull request by Thar0**
This PR adds Makefile machinery for building samples, laying the groundwork for building samplebanks and soundfonts. Note that samples are not yet built, as nothing currently depends on these rules. Using these rules will begin in the next PR where samplebanks will be built.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -213,6 +214,24 @@ $(shell mkdir -p asm data extracted)
...
+ifneq ($(wildcard assets/audio/samples),)
+  SAMPLE_DIRS := $(shell find assets/audio/samples -type d)
+else
+  SAMPLE_DIRS :=
+endif
+
+SAMPLE_FILES         := $(foreach dir,$(SAMPLE_DIRS),$(wildcard $(dir)/*.wav))
+SAMPLE_EXTRACT_FILES := $(foreach dir,$(SAMPLE_EXTRACT_DIRS),$(wildcard $(dir)/*.wav))
+AIFC_FILES           := $(foreach f,$(SAMPLE_FILES),$(BUILD_DIR)/$(f:.wav=.aifc)) $(foreach f,$(SAMPLE_EXTRACT_FILES:.wav=.aifc),$(f:$(EXTRACTED_DIR)/%=$(BUILD_DIR)/%))
+SAMPLE_BLOBS_IN      := $(foreach dir,$(SAMPLE_EXTRACT_DIRS),$(wildcard $(dir)/*.bin))
```
Comment (hensldm) [inline Makefile:232]: What does the `IN` mean? Perhaps `BIN` is better since it seems to find `bin` files?

**Review by hensldm — COMMENTED** [review]
1 MM specific question.

File: Makefile (hunk 1 above)
Comment (Thar0) [reply Makefile:232]: It's meant to be short for "input" since it's the input to the build that's written during extraction

**Review by hensldm — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -505,6 +524,30 @@ $(BUILD_DIR)/assets/%.jpg.inc.c: assets/%.jpg
 $(BUILD_DIR)/%.schl.inc: %.schl
 	$(SCHC) $(SCHC_FLAGS) -o $@ $<
 
+# Audio
+
+AUDIO_BUILD_DEBUG ?= 0
```
Comment (AngheloAlf) [inline Makefile:529]: I wouldn't mind moving this to the top of the Makefile with the other settings we have, but since this is kinda niche I guess is fine to have it here

**Review by AngheloAlf — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -505,6 +524,30 @@ $(BUILD_DIR)/assets/%.jpg.inc.c: assets/%.jpg
...
+
+# first build samples...
+
+$(BUILD_DIR)/assets/audio/samples/%.half.aifc: assets/audio/samples/%.half.wav
+	$(SAMPLECONV) vadpcm-half $< $@
+
+$(BUILD_DIR)/assets/audio/samples/%.half.aifc: $(EXTRACTED_DIR)/assets/audio/samples/%.half.wav
+	$(SAMPLECONV) vadpcm-half $< $@
+ifeq ($(AUDIO_BUILD_DEBUG),1)
```
Comment (AngheloAlf) [inline Makefile:538]: Most of the makefile does this kind of checks against 0 instead, so the consistency would be nice
```suggestion
ifneq ($(AUDIO_BUILD_DEBUG),0)
```

File: Makefile (hunk 2 above)
Comment (Thar0) [reply Makefile:529]: I think I'd like to leave it here for now at least, it can always be moved later

File: Makefile (hunk 3 above)
Comment (Thar0) [reply Makefile:538]: I think this is less readable and there's already `ifeq ($(ORIG_COMPILER),1)`, and in oot's makefile there's `ifeq ($(DEBUG),1)`

Comment (hensldm) [reply Makefile:538]: Fwiw, the main reason to use `ifneq` 0 is so that `0` means OFF while everything else means ON instead of `ifeq` 1 means `1` is ON and everything else is off.

Comment (AngheloAlf) [reply Makefile:538]: Currently, out of the 10 boolean-like checks we have on this makefile, and only 1 is checked against `1` instead of against `0`.

In OoT there are 14 boolean checks, and 7 of them are checks against `1`.

So MM has a majority and OoT is half and half. I do think comparing against `0` is better since it is more boolean-like.

Comment (fig02) [reply Makefile:538]: double negative in general is really hard to parse. Im not a fan of `ifndef` in general, even for things like NDEBUG.
I have no horse in this race but think that how tharo has it is much more readable
